identity-iota/src/document/resolved_iota_document.rs

```diff
@@ -101,3 +101,46 @@ impl From<IotaDocument> for ResolvedIotaDocument {
     }
   }
 }
+
+impl AsRef<IotaDocument> for ResolvedIotaDocument {
+  fn as_ref(&self) -> &IotaDocument {
+    &self.document
+  }
+}
+
+#[cfg(test)]
+mod tests {
+
+  use identity_core::convert::FromJson;
+  use identity_core::convert::ToJson;
+  use identity_core::crypto::KeyPair;
+  use identity_core::crypto::KeyType;
+  use identity_core::crypto::PrivateKey;
+  use identity_core::crypto::PublicKey;
+
+  use super::*;
+
+  // Characterization test: We need to be informed if it becomes impossible to deserialize a serialized `IotaDocument`
+  // into a `ResolvedIotaDocument` as the Wasm bindings currently depend on this fact.
+  #[test]
+  fn can_deserialize_from_iota_document() {
+    let private_key: Vec<u8> = vec![
+      40, 185, 109, 70, 134, 119, 123, 37, 190, 254, 232, 186, 106, 48, 213, 63, 133, 223, 167, 126, 159, 43, 178, 4,
+      190, 217, 52, 66, 92, 63, 69, 84,
+    ];
+    let public_key: Vec<u8> = vec![
+      212, 151, 158, 35, 16, 178, 19, 27, 83, 109, 212, 138, 141, 134, 122, 246, 156, 148, 227, 69, 68, 251, 190, 31,
+      25, 101, 230, 20, 130, 188, 121, 196,
+    ];
+    let keypair = KeyPair::from((
+      KeyType::Ed25519,
+      PublicKey::from(public_key),
+      PrivateKey::from(private_key),
+    ));
```

Comment:
Only the private key is necessary if you use `try_from_ed25519_bytes`. You can also just use a private key of all zeroes rather than an explicit private key, e.g. `[0; 32]`.